### DISCUSSION TITLE
Create msi packages in Windows NanoServer docker container

### DISCUSSION THREAD

**Discussion by yodasad** [wixtoolset/issues#7596] [Questions] [answered]
We use docker containers to build different distributable packages of our product in a CI pipeline.
Now we plan to support MSI packages and want to use Windows NanoServer docker container with .NET SDK for this purpose.
Since NanoServer is missing MSI capabilities building a WIX project resulting in an error

```
wix.exe : error WIX0001: System.DllNotFoundException: Unable to load DLL 'msi.dll' or one of its dependencies: The specified procedure could not be found. (0x8007007F) [C:\proj\TestInstaller.wixproj]
```

Our package is simply built from files in a directory (we use WixToolset.Heat package), we do not need UI, registry, COM and other features.

Is there any way to build an MSI package using WIX in NanoServer or using Server Core image is the only option?

**Comment by robmen** [comment, marked as answer]
`msi.dll` is a prerequisite for WiX to build installation packages for the Windows Installer today. That is part of Windows normally so you'll want to see if/how to add it to Nano Server.

**Comment by yodasad** [comment]
Thank you for the answer!
Unfortunately, it is not so easy to add `msi.dll` to NanoServer, it seems like it has many other dependencies, some of them (for example, `ole32.dll`) breaks the `dotnet.exe` once being added to `C:\Windows\System32\`.

I continue my experiments in the background, and we will use Windows Server Core images for our CI builds.
